torch/nn/modules/upsampling.py

```diff
@@ -64,30 +45,48 @@ class UpsamplingNearest2d(_UpsamplingBase):
         [torch.FloatTensor of size 1x1x4x4]
 
     """
+    def __init__(self, size=None, scale_factor=None):
```

Comment:
It'd be the best to refactor the checks into a single function that could be called both in `_functions` and in here. Right now the code is duplicated.